### PR TITLE
added try excepts for unhandled exceptions

### LINKED ISSUE #776: Unhandled exceptions occurring inside function passed to `executor.map` are swallowed
Currently, if an unhandled exception occurs inside [`convert_doc`](https://github.com/freedomofpress/dangerzone/blob/7c4e62954f4ee71726de0a23655c8f791f05f290/dangerzone/logic.py#L75) when running `dangerzone-cli`, the exception is swallowed (traceback is not logged, the exit code is not changed).

This manifests in tests with dummy conversion - e.g. if [`shutil.copy`](https://github.com/freedomofpress/dangerzone/blob/7c4e62954f4ee71726de0a23655c8f791f05f290/dangerzone/isolation_provider/dummy.py#L65) raises an exception (e.g. `src` and `dst` being the same path), the `CLIResult` is still reported as a success due to the swallowed exception. [Here](https://github.com/naglis/dangerzone/commit/0d214c14d3c9045b9eeb7764fdedb55df4d9a5a3#diff-4e8715c7a425ee52e74b7df4d34efd32e8c92f3e60bd51bc2e1ad5943b82032eR222) is a demo test case.

IIUC during actual (not dummy) execution it would hide potential exceptions occurring [here](https://github.com/freedomofpress/dangerzone/blob/7c4e62954f4ee71726de0a23655c8f791f05f290/dangerzone/isolation_provider/base.py#L69-L70) or [here](https://github.com/freedomofpress/dangerzone/blob/7c4e62954f4ee71726de0a23655c8f791f05f290/dangerzone/isolation_provider/base.py#L90-L94).

I believe this is because the generator returned by `executor.map` is not consumed -  from [`ThreadPoolExecutor` docs](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Executor.map):

> If a _fn_ call raises an exception, then that exception will be raised when its value is retrieved from the iterator.

Here is a simplified form showing the issue:

```python
>>> from concurrent.futures import ThreadPoolExecutor
>>>
>>> def do_work(i):
...     print(f"do_work({i})")
...     1 / 0
...
>>> with ThreadPoolExecutor(max_workers=4) as executor:
...     _ = executor.map(do_work, range(4))
...
do_work(0)
do_work(1)
do_work(2)
do_work(3)
>>>
```

### PR DESCRIPTION
**Pull request by amnak613**
### Description
-what: added a try catch to the areas in the code mentioned in the pr
-why: will help any unhandled errors that are being swallowed within the code

### Test Case

- [x] Ensure all tests pass

Closes #776

### CODE REVIEW

File: dangerzone/isolation_provider/base.py
Comment (apyrgio) [inline dangerzone/isolation_provider/base.py:95]: Adding a try...except block here is not necessary, since the plan is to catch exceptions even higher up. I would ditch this change.

File: dangerzone/logic.py
Comment (apyrgio) [inline dangerzone/logic.py:80]: ... but not that high up. The original issue is that `executor.map()` swallows exceptions by design. This means that we don't gain something by wrapping `executor.map()` in a try...except block. Instead, we need to catch exceptions within the `convert_doc()` function, which is the highest an exception from the conversion may reach.

Something like :

```suggestion
      def convert_doc(document: Document) -> None:
          try
                self.isolation_provider.convert(
                    document,
                    ocr_lang,
                    stdout_callback,
                )
            except Exception:
                log.exception(f"Unexpected error occurred while converting '{document}'")
                document.mark_as_failed()
```

**Review by apyrgio — CHANGES_REQUESTED** [review]
Hey @amnak613! Thanks a lot for your first contribution in the project. It seems that the CI tests are failing, so I've suggested a way to improve your solution and make them work. I think once you fix it, we'll be good to merge.

File: dangerzone/logic.py
Comment (amnak613) [reply dangerzone/logic.py]: Thanks I updated the code so it is contained within the convert_doc function and the unit tests are passing

File: dangerzone/isolation_provider/base.py
Comment (amnak613) [reply dangerzone/isolation_provider/base.py]: Removed this change

**Comment by amnak613** [comment]
@apyrgio All tests are passing and was able to resolve some of the linter errors. I'm having some issues trying to figure out how to get rid of the last lint error on test_cli.py

File: tests/test_cli.py
Comment (almet) [inline tests/test_cli.py:221]: nitpick: you can remove this comment

**Review by almet — COMMENTED** [review]
Thanks for this contribution, this is appreciated!

About the linting error you see, you should be able to fix it by running `make format` (which will apply `isort`, which reorders the imports alphabetically).

We changed the name of this utility recently, so be sure to get the latest changes from the `main` branch by running `git rebase main`.

This should also make the CI happier about Fedora 40 and Debian Trixie, [which we fixed last week](#914)

**Review by apyrgio — APPROVED** [review]

**Comment by apyrgio** [comment]
Thanks a lot for the changes @amnak613. I agree with @almet, with some minimal changes the PR will be good to merge. If you want to, we can make those ourselves and just merge the PR.

File: tests/test_cli.py
Comment (amnak613) [reply tests/test_cli.py:221]: I'll fix this up, I can go ahead and add the changes and then merge the pr

**Comment by amnak613** [comment]
@apyrgio @almet Updated and added changes once the pr has all the approvals will merge

**Review by apyrgio — APPROVED** [review]

**Comment by apyrgio** [comment]
Squashed the commits and verified that the CI passed when the branch was last updated (now the CI fails, but that's due to a CircleCI brown-out). Thanks for the changes @amnak613.

**Pull request merged**
